Code/GraphMol/SLNParse/SLNParse.cpp

```diff
@@ -178,11 +178,11 @@ namespace RDKit {
     };
   } // end of SLNParse namespace
   
-  RWMol *SLNToMol(std::string sln,bool sanitize,int debugParse){
+  RWMol *SLNToMol(const std::string &sln,bool sanitize,int debugParse){
     // FIX: figure out how to reset lexer state
     yysln_debug = debugParse;
     // strip any leading/trailing whitespace:
-    boost::trim_if(sln,boost::is_any_of(" \t\r\n"));
+    //boost::trim_if(sln,boost::is_any_of(" \t\r\n"));
```

Comment:
Good catch, I'll have to fix the SLN parser, the trim_if was moved into the
smiles/smarts lexer.

On Fri, Sep 25, 2015 at 5:50 PM, Greg Landrum <notifications@github.com>
wrote:

> In Code/GraphMol/SLNParse/SLNParse.cpp
> <https://github.com/rdkit/rdkit/pull/627#discussion_r40479184>:
>
> >      // FIX: figure out how to reset lexer state
> >      yysln_debug = debugParse;
> >      // strip any leading/trailing whitespace:
> > -    boost::trim_if(sln,boost::is_any_of(" \t\r\n"));
> > +    //boost::trim_if(sln,boost::is_any_of(" \t\r\n"));
>
> This changes the behavior of the parser, right?
> Any particular reason to do this aside from the fact that you have to
> since the function takes a const?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rdkit/rdkit/pull/627/files#r40479184>.
>